pkg/oc/bootstrap/clusterup/kubeapiserver/openshift_controller.go

```diff
@@ -0,0 +1,45 @@
+package kubeapiserver
+
+import (
+	"io/ioutil"
+	"path"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/golang/glog"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/apis/config/latest"
+	"github.com/openshift/origin/pkg/oc/bootstrap/clusterup/tmpformac"
+)
+
+func MakeOpenShiftControllerConfig(existingMasterConfig string, basedir string) (string, error) {
+	configDir := path.Join(basedir, OpenShiftControllerManagerDirName)
+	glog.V(1).Infof("Copying kube-apiserver config to local directory %s", OpenShiftControllerManagerDirName)
+	if err := tmpformac.CopyDirectory(existingMasterConfig, configDir); err != nil {
+		return "", err
+	}
+
+	// update some listen information to include starting the DNS server
+	masterconfigFilename := path.Join(configDir, "master-config.yaml")
+	originalBytes, err := ioutil.ReadFile(masterconfigFilename)
+	if err != nil {
+		return "", err
+	}
+	configObj, err := runtime.Decode(configapilatest.Codec, originalBytes)
+	if err != nil {
+		return "", err
+	}
+	masterconfig := configObj.(*configapi.MasterConfig)
+	masterconfig.ServingInfo.BindAddress = "0.0.0.0:8444"
```

Comment:
do we want to bind this on detected IP addresses?

Reply:
> do we want to bind this on detected IP addresses?

we don't do that for the rest.  I think I'll go with "all or nothing"